src/ape_ethereum/trace.py

```diff
@@ -0,0 +1,538 @@
+            if stylize
+            else _list_to_str(outputs)
+        )
+
+    return f"[{TraceStyles.OUTPUTS}]{outputs}[/]" if stylize else str(outputs)
+
+
+def _dict_to_str(dictionary: Dict, color: Optional[str] = None) -> str:
+    length = sum(len(str(v)) for v in [*dictionary.keys(), *dictionary.values()])
+    do_wrap = length > _WRAP_THRESHOLD
+
+    index = 0
+    end_index = len(dictionary) - 1
+    kv_str = "(\n" if do_wrap else "("
+
+    for key, value in dictionary.items():
+        if do_wrap:
+            kv_str += _INDENT * " "
+
+        if isinstance(value, (list, tuple)):
+            value = _list_to_str(value, 1 if do_wrap else 0)
+
+        value_str = f"[{color}]{value}[/]" if color is not None else str(value)
+        kv_str += f"{key}={value_str}" if key and not key.isnumeric() else value_str
+        if index < end_index:
+            kv_str += ", "
+
+        if do_wrap:
+            kv_str += "\n"
+
+        index += 1
+
+    return f"{kv_str})"
+
+
+def _list_to_str(ls: Union[List, Tuple], depth: int = 0) -> str:
+    if not isinstance(ls, (list, tuple)) or len(str(ls)) < _WRAP_THRESHOLD:
+        return str(ls)
+
+    elif ls and isinstance(ls[0], (list, tuple)):
+        # List of lists
+        sub_lists = [_list_to_str(i) for i in ls]
+
+        # Use multi-line if exceeds threshold OR any of the sub-lists use multi-line
+        extra_chars_len = (len(sub_lists) - 1) * 2
+        use_multiline = len(str(sub_lists)) + extra_chars_len > _WRAP_THRESHOLD or any(
+            ["\n" in ls for ls in sub_lists]
+        )
+
+        if not use_multiline:
+            # Happens for lists like '[[0], [1]]' that are short.
+            return f"[{', '.join(sub_lists)}]"
+
+        value = "[\n"
+        num_sub_lists = len(sub_lists)
+        index = 0
+        spacing = _INDENT * " " * 2
+        for formatted_list in sub_lists:
+            if "\n" in formatted_list:
+                # Multi-line sub list. Append 1 more spacing to each line.
+                indented_item = f"\n{spacing}".join(formatted_list.splitlines())
+                value = f"{value}{spacing}{indented_item}"
+            else:
+                # Single line sub-list
+                value = f"{value}{spacing}{formatted_list}"
+
+            if index < num_sub_lists - 1:
+                value = f"{value},"
+
+            value = f"{value}\n"
+            index += 1
+
+        value = f"{value}{_INDENT * ' '}]"
+        return value
+
+    return _list_to_multiline_str(ls, depth=depth)
+
+
+def _list_to_multiline_str(value: Union[List, Tuple], depth: int = 0) -> str:
+    spacing = _INDENT * " "
+    new_val = "[\n"
+    num_values = len(value)
+    for idx in range(num_values):
+        ls_spacing = spacing * (depth + 1)
+        new_val += f"{ls_spacing}{value[idx]}"
+        if idx < num_values - 1:
+            new_val += ","
+
+        new_val += "\n"
+
+    new_val += spacing * depth
+    new_val += "]"
+    return new_val
```

Comment:
```suggestion
    return (
        "[\n"
        + f",\n{spacing * (depth + 1)}".join(str(v) for v in value)
        + f"\n{spacing * depth}]"
    )
```

Reply:
Thanks! I had to make 1 slight adjustment for accuracy:

```python
  spacing = _INDENT * " "
  ls_spacing = spacing * (depth + 1)
  joined = ",\n".join([f"{ls_spacing}{v}" for v in value])
  new_val = f"[\n{joined}\n{spacing * depth}]"
  return new_val
```